tasks/sync-granule/GranuleFetcher.js

```diff
@@ -183,6 +161,69 @@ class GranuleFetcher {
       file.name.match(fileConfig.regex));
   }
 
+  /**
+   * Check whether file object qualifies as a checksum file based on
+   * file extension or file collection config.
+   *
+   * @param {Object} file - file object
+   * @returns {boolean} - whether file is a checksum file
+   * @private
+   */
+  isChecksumFile(file) {
+    return this.hasChecksumForFileConfig(file) || ['.md5', '.cksum', '.sha1', '.sha256'].includes(path.extname(file.name));
+  }
+
+  /**
+   * Determine whether given file has a checksumFor property
+   *
+   * @param {Object} file - file object
+   * @returns {boolean} - true/false collection config for file has checksumFor property
+   * @private
+   */
+  hasChecksumForFileConfig(file) {
+    return has(
+      this.findCollectionFileConfigForFile(file),
+      'checksumFor'
+    );
+  }
+
+  /**
+   * Add checksum types and values to files
+   *
+   * @param {Array<Object>} files - files objects with name and path
+   * @returns {Array<Object>} - files with checksum types and values added
+   */
+  async addChecksumsToFiles(files) {
```

Comment:
I'd be happy to pair on it, or take a swing at it myself, but this function is pretty complex and has a lot of room for hiding bugs. Feel free to say, "alright, then, _you_ refactor it."

Reply:
I've been looking at this for like a day and not seeing any obvious way to simplify or clean it up, so I'm gonna pass on the opportunity. 😅

Reply:
That's moved beyond "code smell" into "code stink" :-(